systemd basics

Amrita42 opened the pull request:
### Description

Describe the overall goals of this pull request.
The scope is to merge/add content/implement feedback to the following:

 Basic systemd concepts and systemd basics

**Important**
[4/10/2023] Based on review feedback , The **connection accounting utilities** will not be covered in this article/PR . I will work on it separately as it needs more work and also as a stand alone article.
[8/10/2023] Termination signals and Creating a systemd service will not be covered in this article , it will be in a separate article.

tomschr reviewed (changes requested):
@Amrita42 Thanks for all your efforts! :hugs: 

I've added some comments, suggestions, and fixing some typos. Apart from these, I also have some general recommendations:

* Prompts
  Some starts with a command, others with a `&prompt...;` entity. Maybe go through all screens and add some prompts where necessary to make them consistent.
* Screens
  Screens must not be indented. :wink: Always start your text in column 1, although it looks bad from a formatting perspective. The reason for this is, this makes screens look inconsistent in _the output_. It looks weird and the user may be asking if the content should also be indented.
  Another reason: your available space may be limited, especially inside some lists. You will waste valuable characters that are probably needed on the right site.

Apart from the above issues, some other points:

* As I've mentioned it below, the section "About system usage" looks a bit odd to me. It seems not really related to systemd or I'm too dumb to see the connection. :laughing: 
* The section "Structure of a unit file" is very general. Maybe it would be more useful to the reader, if you use an existing (simple) unit file and try to dissect it and explain the necessary pieces?
* The section "Creating a Linux service with systemd" explains how to create the service file, but it actually doesn't explain the `FILE_NAME.sh`. That's the thing that gets started when the service is triggered. What does it contain? What does it do? This is missing.
  Maybe it would be more helpful to have a specific example?

taroth21 reviewed (changes requested):
Hi @Amrita42, many thanks - I'll attach the first part of my review as annotated PDF to this PR and will finish the review at the beginning of next week. 

[what-is-systemd_feedback_taroth_part1.pdf](https://github.com/SUSE/doc-modular/files/12767663/what-is-systemd_feedback_taroth_part1.pdf)

taroth21 commented:
Hi Amrita, please find my feedback for the rest of the article here:
[what-is-systemd_feedback_taroth_part2.pdf](https://github.com/SUSE/doc-modular/files/12781534/what-is-systemd_feedback_taroth_part2.pdf)

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,67 @@
+      <meta name="maintainer" content="amrita.sakthivel@suse.com" its:translate="no"/>
+    <abstract>
+     <para>
+      Administration is easier when you understand units and unit files when working with &systemd;.
```
Thanks Toms :) I will use "unit file"

Amrita42 commented on concepts/types-of-systemd-unit-files.xml:
```diff
@@ -0,0 +1,148 @@
+      <meta name="maintainer" content="amrita.sakthivel@suse.com" its:translate="no"/>
+    <abstract>
+     <para>
+      You can determine the type of a unit with the <literal>Type=</literal> suffix which is appended to
+      the end of the resource name. &systemd; categorizes units according to the type of resource they describe.
```
Good point

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,67 @@
+      <term>[Install]</term>
+      <listitem>
+        <para>
+         This is often the last section in the unit file and is optional. This section is used to define the behavior of a unit when
+         it is enabled or disabled. When you enable a unit, it automatically starts at boot. </para>
```
@tomschr its not clear , do you mean the directives in the install section or add as a sentence ?

tomschr replied:
I'm sorry, @Amrita42. I was referring to the dependencies of the unit itself. Maybe there is a better term for this. For example, the Crony service has these "dependencies" in the `Unit` section:

```INI
[Unit]
Description=NTP client/server
Documentation=man:chronyd(8) man:chrony.conf(5)
After=nss-lookup.target
Wants=network.target
After=network.target
Wants=time-sync.target
Before=time-sync.target
Conflicts=ntpd.service systemd-timesyncd.service
```

Here, the the "After", "Wants", and "Before" are all "dependencies" that the Crony service needs as a requisites. There are more of these keywords. 

My idea was to just mention in a sentence that a unit may require other service types to work properly. Roughly speaking, something like this:

```
Depending on the unit it may require other units to work properly. For example, ...
```

Does that make sense?

Amrita42 replied:
Yes , its very clear , I have added a sentence.

Amrita42 commented on concepts/types-of-systemd-unit-files.xml:
```diff
@@ -0,0 +1,148 @@
+    </abstract>
+      </info>
+
+  <para>Types of unit files available for &systemd;:</para>
```
re-phrased

Amrita42 commented on concepts/types-of-systemd-unit-files.xml:
```diff
@@ -0,0 +1,148 @@
+      <listitem>
+        <para>
+          A service unit file describes how to manage a service or application on the server.
+          This includes how to start or stop the service,under what conditions the service starts automatically,
+          and the dependency or the hierarchy information for related software.
```
let me re-phrase , dependency on the other unit files is what i meant

Amrita42 commented on tasks/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,62 @@
+      </step>
+      <step>
+        <para>
+          Create a &systemd; unit file and save it at <filename>~/.config/systemd/user/</filename> directory:
```
Its explained below i.e. FILE_NAME.service

Amrita42 commented on tasks/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,62 @@
+      <step>
+        <para>
+          You can manage &systemd; services as a regular user with the <literal>user</literal>
+          option of the <command>systemctl</command> command. As a regular user, you must provide the path to the unit files
+         otherwise if a unit file with the same name exists, it would be executed or listed instead.
```
I had a similar question when I looked at this [section 8 of systemdtimers](https://documentation.suse.com/smart/systems-management/html/systemd-working-with-timers/index.html#systemd-timer-user) >> As a regular user, you must provide the path to the unit files as in the examples below. Otherwise, if a system-wide timer with the same name exists, it would be executed or listed instead.
@fsundermeyer suggested  specifying the complete path ensures there are are no errors otherwise how would the user know if a systemd file with same name exists

fsundermeyer replied:
What Amrita said. In _this_ particular  case the name of the unit file is unique, so not specifying the path works. But if a system unit file and a user unit file have the same name, the system file takes precedence. As a user you are probably not awayre of all existing system unit files, so it is a good practice to _always_ specify a path when running systemd as a user.

Amrita42 commented on concepts/types-of-systemd-unit-files.xml:
```diff
@@ -0,0 +1,148 @@
+        The <literal>.path</literal> unit file defines a path for path based activation.
+        By default, a <literal>.service</literal> unit file of the same base name is started
+        when the path reaches a specified state.
+        <literal>inotify</literal> is used to monitor the path for changes.
```
Linux has used inotify to monitor for changes to paths and files. The inotify kernel API can be used by programs that want to be notified about changes to files

Amrita42 replied:
re-phrased to define

Amrita42 commented:
@tomschr 
As I've mentioned it below, the section "About system usage" looks a bit odd to me. It seems not really related to systemd or I'm too dumb to see the connection. 😆
[Amrita] it is not directly related but i feel useful info to have .
The section "Creating a Linux service with systemd" explains how to create the service file, but it actually doesn't explain the FILE_NAME.sh. That's the thing that gets started when the service is triggered. What does it contain? What does it do? This is missing.
Maybe it would be more helpful to have a specific example?
[Amrita]: I have explained this in _concepts/systemd-unit-file.xml_  , the anatomy of the unit file. I thought it is more specific to creating the a systemd service. Maybe add a link for additional resource or to know more about a unit file see...

Amrita42 commented on references/systemd-gnu-accounting-utilities.xml:
Removed

Amrita42 commented:
@taroth21 Addressed the feedback from part 1 and 2. Please note for the systemctl commands , I have added example output/also expanded decsriptions where applicable because start/enable/stop etc does not have any output .

Amrita42 commented:
**Pending/wip**: two example outputs are nor rendering properly in **systemctl-cheatsheet.xml**

fsundermeyer reviewed (changes requested):
Amrita, I think this article needs to be stripped down further. If I understand correctyl it's primary goal is to provide an introduction on the baseics of systemd. Goal of this article is to help readers to understand what systemd does and where a user can interact with systemd.
Everything else (e.g. How to create your own services, How to interprete the output of a specific systemctl command, etc) is subject for aother articles on the respective subject.
Once we have created these specific articles, this article here should have pointers to these specific articles (e.g. the systemd timers article).

Therefore I would remove some sections (and re-use them in the more specific articles) . I would also add (small sections or at least some pointers to):
* the systemd concept of logging (only the concept, everything else would be covered in a seperate article about journalctl)
* dependency handling between units(again, concept only, the details are subject to the creating my own service article)
* an extra section about the concept of targets (as a mechanism to group services)

Amrita42 commented on articles/systemd-services.asm.xml:
```diff
@@ -0,0 +1,193 @@
+  <!-- S T R U C T U R E -->
+  <structure renderas="article" xml:id="systemd-services" xml:lang="en">
+    <merge>
+      <title>Managing &systemd; services</title>
```
Re-phrased as systemd basics

Amrita42 commented on DC-what-is-systemd:
```diff
@@ -0,0 +1,18 @@
+STYLEROOT="/usr/share/xml/docbook/stylesheet/suse2022-ns"
+FALLBACK_STYLEROOT="/usr/share/xml/docbook/stylesheet/suse2021-ns"
+
+DOCBOOK5_RNG_URI="urn:x-suse:rng:v2:geekodoc-flat"
```
good catch

Amrita42 commented on references/systemd-termination-signal-new.xml:
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
```
Yes this was mentioned in Tanja's part 2 pdf review and I have moved the file >> systemd-termination-signal-new.xml. But based on your review , I have removed this ref topic for a new article inclusion

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,95 @@
+ xmlns:xlink="http://www.w3.org/1999/xlink"
+ xmlns:trans="http://docbook.org/ns/transclusion">
+  <info>
+    <title>Structure of a unit file</title>
```
Since we are explaining the anatomy of a unit file , I would prefer to use "structure of a unit file"

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,95 @@
+      Administration is easier when you understand unit files when working with &systemd;.
+     Unit files use a simple declarative syntax that allows you to see easily,
+     the purpose and effects of a unit upon activation. Unit files have sections with directives, for example:
+      </para>
```
Done

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,95 @@
+        <para>
+          The <literal>[Service]</literal> section provides configuration that is only applicable for services.
+          You must specify <literal>Type=</literal> in the <literal>[Service]</literal> section. This categorizes services
+          by their process and daemon behavior. It instructs  &systemd; how to correctly manage the service and also determine its state.
```
To give context, I had a dilemma whether toi includes all the sections ( would it be too much?) or part of it so I went with the line and listed the most common/imp ones :
_Unit files include the following sections:_

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,95 @@
+         This is often the last section in the unit file and is optional. This section is used to define the behavior of a unit file when
+         it is enabled or disabled. When you enable a unit file, it automatically starts at boot. </para>
+              </listitem>
+    </varlistentry>
```
To give context, I had a dilemma whether toi includes all the sections ( would it be too much?) or part of it so I went with the line and listed the most common/imp ones :
_Unit files include the following sections:_

Amrita42 replied:
@fsundermeyer

fsundermeyer replied:
How about 

[Service] / [Mount] / [Path] / ...
Unit-type-specific section providing configuration only applicable for the respective type. For the mount type, for example, this section specifies mount points and the file system to mount. For the service it would be information on which daemon to start.

Amrita42 replied:
Added [Mount] and [Path] , [[Service] is defined

fsundermeyer commented on concepts/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,61 @@
+  <section xml:id="systemd-regular">
+    <title>Introduction</title>
+    <para>
+      By default, &systemd; runs with &rootuser; user privileges. In certain scenarios it is more secure
```
I am not sure the security argument is legit here. I'd rather say you can also use systemd when you do not have root privileges. Or when you want/need to start services as a user. Or when you want to automate certain processes (via systemd timers).

Amrita42 replied:
re-phrased

fsundermeyer commented on concepts/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,61 @@
+      service.
+      </para>
+  </section>
+  <section xml:id="create-user-service">
```
Again, no subsection here.

I also think this section is too specific, we will have a separate article on this topic where we can provide all the details. Think of this part as the introduction for that article. So rather explain that systemd _can_ be used without root privileges and why you want to do this.

Amrita42 replied:
Implemented.

fsundermeyer commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,269 @@
+            <command>--state=running</command> to show the active units and <command>--type=service</command> to
+            show the exited and active units.
+          </para>
+          <para><emphasis>Example output</emphasis></para>
```
No examples in this section, please remove all of them

Amrita42 replied:
Done

Amrita42 commented on articles/systemd-basics.asm.xml:
```diff
@@ -0,0 +1,194 @@
+            <term>GOAL</term>
+            <listitem>
+              <para>
+              You can manage all system settings and services with &systemd;.
```
Removed GOAL

Amrita42 commented on concepts/what-is-systemd.xml:
```diff
@@ -0,0 +1,44 @@
+          Target unit files have a <literal>.target</literal> file extension. Target units group together various &systemd;
+          units through a chain of dependencies.
+        </para>
+        <para>For troubleshooting, you can use <command>systemctl</command> and <command>journalctl</command>, which is
```
Implemented

Amrita42 commented on concepts/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,61 @@
+    <title>Using &systemd; as a regular user</title>
+    <meta name="maintainer" content="amrita.sakthivel@suse.com" its:translate="no"/>
+  </info>
+  <section xml:id="systemd-regular">
```
Done

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,119 @@
+        </callout>
+        <callout arearefs="systemd-unit-file-multiuser">
+          <para>
+            The most common way to specify how a service should be enabled.
```
re-defined

Amrita42 commented:
@dariavladykina Please provide a peer/style review . Thanks

dariavladykina commented on concepts/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,45 @@
+  <info>
+    <title>Using &systemd; as a regular user</title>
+    <meta name="maintainer" content="amrita.sakthivel@suse.com" its:translate="no"/>
+<abstract><para> You can use &systemd; as a regular user and when you don't have &rootuser; user privileges.
+  In certain scenarios it is more secure for a service to run with the privilege of a regular user. You can do this by creating a <literal>user</literal>
```
```suggestion
<abstract><para> You can use &systemd; as a regular user and when you do not have &rootuser; user privileges.
  In certain scenarios, it is more secure for a service to run with the privilege of a regular user. You can do this by creating a <literal>user</literal>
```

dariavladykina commented on concepts/systemd-as-regular-user.xml:
```diff
@@ -0,0 +1,45 @@
+      <listitem>
+        <para>
+         User service sessions are terminated when the user's session ends. This can be overridden by using
+          the <command>loginctl enable-linger <replaceable>USERNAME</replaceable></command>
```
```suggestion
          the <command>loginctl enable-linger <replaceable>USERNAME</replaceable></command>.
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+        <term>systemctl list-units</term>
+        <listitem>
+          <para>
+            Lists the &systemd; units. You can use the optional arguments;
```
```suggestion
            Lists the &systemd; units. You can use the optional arguments:
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+  <term>systemctl start <replaceable>SERVICE</replaceable></term>
+<listitem>
+  <para>
+    Instead of manually starting the service , use the <command>start</command>command.
```
```suggestion
    Instead of manually starting the service, use the <command>start</command> command.
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+  <term>systemctl disable <replaceable>SERVICE</replaceable></term>
+<listitem>
+  <para>
+    Disables the service at boot.
```
```suggestion
    Disables the service on boot.
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+<listitem>
+  <para>
+    Reload the service if it supports reloading, otherwise it restarts the service.
+    If the service is not running, it is restarted .
```
```suggestion
    If the service is not running, it is restarted.
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+  <para>
+    When a service is masked, this means the unit file is symlinked to <filename>/dev/null</filename>.
+  A symlink for a masked service is created from <filename>/etc/systemd/system</filename> to point to <filename>/dev/null</filename>.
+  This makes it impossible to load the service even if another enabled service requires it. It must be stopped manually or
```
```suggestion
  This makes it impossible to load the service even if another enabled service requires it. It must be stopped manually, or
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+  <term>systemctl unmask <replaceable>SERVICE</replaceable></term>
+<listitem>
+  <para>
+   Unmasks the service. It is effective when the system restarts, or starts manually.
```
```suggestion
   Unmasks the service. It is effective when the system restarts or starts manually.
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+    </varlistentry>
+    <varlistentry>
+      <term>systemctl emergency</term>
+      <listitem><para>Goes into the emergency mode<literal>emergency.target</literal>.</para></listitem>
```
```suggestion
      <listitem><para>Goes into the emergency mode <literal>emergency.target</literal>.</para></listitem>
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+      <listitem>
+        <para>
+          If a service is causing issues during the boot process, you can use the
+          <command>systemd-analyze plot command</command> to visual the boot process and identify issues.
```
```suggestion
          <command>systemd-analyze plot command</command> to visualize the boot process and identify issues.
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+        </para>
+
+<screen>&prompt.sudo;systemd-analyze plot > boot-plot.svg</screen>
+<para>The command creates a SVG file called <filename>boot-plot.svg</filename>that contains a graphical representation
+  of the boot process and the potential issues. This includes the start and stop time of each service. You can open this
+file in a SVG compatible image viewer or web browser to analyze services that are causing issues during the boot process.</para>
```
```suggestion
<para>The command creates an SVG file called <filename>boot-plot.svg</filename> that contains a graphical representation of the boot process and potential issues. This includes the start and stop time of each service. You can open this file in an SVG-compatible image viewer or Web browser to analyze services that are causing issues during the boot process.</para>
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+  <variablelist>
+    <varlistentry>
+      <term><emphasis>Check the syntax of your &systemd; unit file with the
+        <command>systemd-analyze verify <replaceable>SERVICE</replaceable></command>.</emphasis></term>
```
```suggestion
        <command>systemd-analyze verify <replaceable>SERVICE</replaceable></command></emphasis></term>
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+       </varlistentry>
+        <varlistentry>
+      <term><emphasis>Check the logs for your service with the <command>
+        journalctl -u <replaceable>SERVICE</replaceable></command> command.</emphasis></term>
```
```suggestion
        journalctl -u <replaceable>SERVICE</replaceable></command> command</emphasis></term>
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+        </listitem>
+    </varlistentry>
+    <varlistentry>
+      <term><emphasis>Use the <command>systemd-analyze plot</command> command to visualize the boot process.</emphasis></term>
```
```suggestion
      <term><emphasis>Use the <command>systemd-analyze plot</command> command to visualize the boot process</emphasis></term>
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+         </varlistentry>
+       <varlistentry>
+       <term><emphasis>Shutdown or reboot takes long</emphasis></term>
+      <listitem> <para>If the shutdown or reboot takes long, it could be a service that is not exiting. &systemd;
+        waits for some time for each service to exit before trying to kill it.
+        A common issue is a suspended service or a stalled shutdown. To find out, use the following:
```
```suggestion
      <listitem> <para>If the shutdown or reboot takes long, it could be a service that is not exiting. &systemd;
        waits for some time for each service to exit before trying to terminate it.
        A common issue is a suspended service or a stalled shutdown. To find out, use the following:
```

dariavladykina commented on tasks/systemd-troubleshooting.xml:
```diff
@@ -0,0 +1,94 @@
+  Failed to power off system via logind: There's already a shutdown or sleep operation in progress
+</screen>
+<screen>&prompt.sudo;systemctl list-jobs</screen>
+<para>You can cancel the running and waiting jobs, and again shutdown or reboot: </para>
```
```suggestion
<para>You can cancel the running and waiting jobs, and again shut down or reboot: </para>
```

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+            <term><emphasis>Use absolute path in your &systemd; unit file</emphasis></term>
+            <listitem>
+              <para>
+               Use absolute path for executable files and required files such as configuration files or scripts in your
```
```suggestion
               Use an absolute path for executable files and required files such as configuration files or scripts in your
```

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+            <term><emphasis>Use the ExecReload directive</emphasis></term>
+            <listitem>
+              <para>
+              Use the <emphasis>ExecReload</emphasis> directive in the <literal>[SECTION]</literal> section, when you want to define a specific command
```
```suggestion
              Use the <emphasis>ExecReload</emphasis> directive in the <literal>[SECTION]</literal> section when you want to define a specific command
```

dariavladykina replied:
Should this be [Service] section instead of [SECTION] section?

Amrita42 replied:
Good catch , yes :)

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+            <term><emphasis>Use the RestartSec directive </emphasis></term>
+            <listitem>
+              <para>
+               Use the <emphasis>ExecReload</emphasis> directive in the <literal>[SECTION]</literal> section, when you want to define
```
Should this be [Service] section instead of [SECTION] section?

Amrita42 replied:
Correct, implemented

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+            <term><emphasis>Use the RestartSec directive </emphasis></term>
+            <listitem>
+              <para>
+               Use the <emphasis>ExecReload</emphasis> directive in the <literal>[SECTION]</literal> section, when you want to define
```
```suggestion
               Use the <emphasis>RestartSec</emphasis> directive in the <literal>[SECTION]</literal> section when you want to
```

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+              <para>
+               Use the <emphasis>ExecReload</emphasis> directive in the <literal>[SECTION]</literal> section, when you want to define
+               define a delay (in seconds) before the service is restarted after a failure. This is useful for services that require
+               a specified time to release resources or to to prevent rapid restart loops that can cause high system load.
```
```suggestion
               a specified time to release resources or prevent rapid restart loops that can cause high system load.
```

Amrita42 replied:
Done

dariavladykina commented on concepts/systemd-best-practices.xml:
```diff
@@ -0,0 +1,84 @@
+            <term><emphasis>Disable emergency mode on a remote machine</emphasis></term>
+            <listitem>
+              <para>
+               You can disable emergency mode on a remote machine for example, a virtual machine hosted on Google Cloud.
+               If this mode is enabled , the machine is blocked from connecting to the network.
```
```suggestion
               You can disable emergency mode on a remote machine, for example, a virtual machine hosted on Google Cloud.
               If this mode is enabled, the machine is blocked from connecting to the network.
```

dariavladykina reviewed (commented):
Hi Amrita, please see some suggestions from my part. We can have another review round when all is settled. Thanks!

Amrita42 commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,104 @@
+      <listitem>
+        <para>
+         This is often the last section in the unit file and is optional. This section is used to define the behavior of a unit file when
+         it is enabled or disabled. When you enable a unit file, it automatically starts at boot. Based on the specific unit,,
+         there could be a dependency on other related units to work properly.For example, the <literal>Crony </literal> requires
+         the directives; <literal>After</literal>, <literal>Wants</literal>, and <literal>Before</literal> are all dependencies for
+         Crony to work.
```
re-phrased >> The Chrony service has the following directives , After,Wants.......

Amrita42 commented on concepts/systemd-targets.xml:
```diff
@@ -0,0 +1,93 @@
+    <meta name="maintainer" content="amrita.sakthivel@suse.com" its:translate="no"/>
+    <abstract>      <!-- can be changed via merge in the assembly -->
+      <para>
+     &systemd; uses units and targets. A &systemd; unit defines a service or action on the system which consists of a name, type, and  configuration
+        file. A &systemd; target combines several units, and defines which services have to be started to reach the target.
+        On a server system,for example, a state where the network is running and multiple users can login.
+        These files are identified by the suffix <literal>.target</literal>
```
It was suggested by Frank in his feedback . But yes , when I check the rest of mod-docs , its just server , so will remove the system word

Amrita42 commented on articles/systemd-basics.asm.xml:
```diff
@@ -0,0 +1,186 @@
+            <term>WHAT?</term>
+            <listitem>
+              <para>
+                Learn about &systemd; basics.
```
Re-phrased to give more details  and yes I have interchanged the places for what and why. 

Thanks Daria

dariavladykina replied:
I cannot see this update in the built doc, unfortunately

Amrita42 replied:
Somehow it was not pushed , now it should reflect.

dariavladykina reviewed (changes requested):
A small batch of suggestions, please.

dariavladykina commented on articles/systemd-basics.asm.xml:
```diff
@@ -0,0 +1,187 @@
+            <term>WHY?</term>
+            <listitem>
+              <para>
+                Learn about &systemd; basics. How it provides essential functionalities such as service management,
+                dependency tracking,logging,resource management,socket activation, and system control. </para>
```
```suggestion
         Learn about the basics of &systemd;, which include essential functionalities such as service management, 
         dependency tracking, logging, resource management, socket activation and system control.
         </para>
```

dariavladykina commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,104 @@
+      <listitem>
+        <para>
+         This is often the last section in the unit file and is optional. This section is used to define the behavior of a unit file when
+         it is enabled or disabled. When you enable a unit file, it automatically starts at boot. Based on the specific unit,,
+         there could be a dependency on other related units to work properly.For example, &chrony; requires
+         the directives; <literal>After</literal>, <literal>Wants</literal>, and <literal>Before</literal>, which are all dependencies for
```
```suggestion
         it is enabled or disabled. When you enable a unit file, it automatically starts at boot. Based on the specific 
         unit, there could be a dependency on other related units to work properly. For example, &chrony; requires
         the directives <literal>After</literal>, <literal>Wants</literal> and <literal>Before</literal>, which are all dependencies for
```

dariavladykina commented on concepts/systemd-unit-dependencies.xml:
```diff
@@ -0,0 +1,66 @@
+        </varlistentry>
+        <varlistentry>
+          <term><literal>After</literal></term>
+          <listitem><para>If a unit A has <literal>After=unit B</literal>, when both units are run, unit B is executed fully
```
```suggestion
          <listitem><para>If unit A has <literal>After=unit B</literal>, when both units are run, unit B is executed fully
```

dariavladykina commented on references/systemctl-cheatsheet.xml:
```diff
@@ -0,0 +1,179 @@
+      </para>
+    </abstract>
+    </info>
+      <para>You can use the following common <command>systemctl</command> common commands and refer to
```
```suggestion
      <para>You can use the following common <command>systemctl</command> commands and refer to
```

dariavladykina commented on concepts/systemd-unit-file.xml:
```diff
@@ -0,0 +1,104 @@
+          Sections containing directives that are specific to the respective type. See
+          <xref linkend="type-unit-file"/>
+          for a list of available types.
+          Note that the types; <literal>device</literal>, <literal>target</literal>, <literal>snapshot</literal>,
+          and <literal>scope</literal> do not have a type-specific section.
```
```suggestion
          Note that the types <literal>device</literal>, <literal>target</literal>, <literal>snapshot</literal>
          and <literal>scope</literal> do not have a type-specific section.
```
